### PR TITLE
Use static memory layout for all sysvars

### PR DESCRIPTION
**Pull request by ripatel-fd**
Annotates some sysvar data structures to guarantee a stable ABI

### CODE REVIEW

File: sdk/program/src/epoch_rewards.rs [hunk 1]
```diff
@@ -7,6 +7,8 @@
 //! [`sysvar::epoch_rewards`]: crate::sysvar::epoch_rewards
 
 use std::ops::AddAssign;
+
+#[repr(C)]
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Default, Clone, Copy, AbiExample)]
 pub struct EpochRewards {
```
Comment (CriesofCarrots) [inline sdk/program/src/epoch_rewards.rs:13]: Just putting this here for completeness: this sysvar will get reworked as soon as https://github.com/solana-foundation/solana-improvement-documents/pull/118 lands, and `repr(C)` is part of the new spec.

**Review by buffalojoec — APPROVED** [review]
Lgtm. 👍

**Review by t-nelson — APPROVED** [review]
good thing these are all full of `u64`s :sweat_smile:

**Comment by 2501babe** [comment]
yep i was going to put repr c on stake history entry if mine would land also

**Review by 2501babe — APPROVED** [review]

**Pull request merged**